content/learn/book/your-first-game/_index.md

```diff
@@ -0,0 +1,685 @@
+            transform: Transform::from_xyz(0.0, bounds.y / 2.0, 0.0),
+            sprite: Sprite::new(Vec2::new(bounds.x + wall_thickness, wall_thickness)),
+            ..Default::default()
+        })
+        .with(Collider::Solid);
+```
+Add this code to the inside of the startup function, just before it ends and run the program with `cargo run`.
+You should now see walls inside of your game just like this:
+
+![](breakout3.png)
+
+With all that we have learned so far, it should be pretty obvious what the code above does. But let's go over some interesting details.
+You might have noticed that except for the last wall, every other wall has received a clone of the wall_material.
+Why is that necessary?  
+Well two fold: It is usually good practice to give every object in a game its own material. Otherwise it can happen, that you change the color on one object and suddenly you
+have changed the color on all objects that share the same material. This is usually not what you want. Secondly, of course Rust would like to have a single material owner per SpriteBundle instance.
+So in this case we can solve both requirements with a simple call to ```wall_material.clone()```.
+
+## Next let's add the Ball. This is where the fun starts :-)
+
+First add this line to our use section at the top:
+```rs
+sprite::collide_aabb::{collide, Collision},
+```
+So it looks like this:
+```rs
+use bevy::{
+    prelude::*,
+    render::pass::ClearColor,
+    sprite::collide_aabb::{collide, Collision},
+};
+```
+This line adds a couple of cool functions that allows us to detect collsions between our ball and the other objects in our game.
+
+Next add these two lines into the middle of our main function at the top:
+```rs
+        .add_system(ball_collision_system.system())
+        .add_system(ball_movement_system.system())
+```
+so it looks like this:
+```rs
+        .add_system(paddle_movement_system.system())
+        .add_system(ball_collision_system.system())
+        .add_system(ball_movement_system.system())
+        .run();
+```
+
+Next add a struct that represents our ball right after the Paddle struct:
+```rs
+struct Ball {
+    velocity: Vec3,
+}
+```
+[Vec3](https://docs.rs/glam/latest/glam/f32/struct.Vec3.html) is a struct that contains three float fields: x,y and z.  
+It is part of the [glam](https://docs.rs/glam/latest/glam/) Rust crate which is a simple and fast linear algebra library for games and graphics.
+
+Add the following code to our setup function, just before we add our walls:
+```rs
+        commands.spawn(SpriteBundle {
+            material: materials.add(Color::rgb(1.0, 0.5, 0.5).into()),
+            transform: Transform::from_xyz(0.0, -50.0, 1.0),
+            sprite: Sprite::new(Vec2::new(30.0, 30.0)),
+            ..Default::default()
+        })
+        .with(Ball {
+            velocity: 400.0 * Vec3::new(0.5, -0.5, 0.0).normalize(),
+        });
+```
+This code should look very familiar by now. ;-)
```

Comment:
```suggestion
This code should look very familiar by now. ;-)
`normalize()` simply sets the magnitude of the direction vector to 1, allowing us to freely change its direction without worrying about how it will impact our ball's speed.
```